components/bl-mapbox-component/src/map.js

```diff
@@ -0,0 +1,232 @@
+import { useEffect, useRef, useState } from 'react';
+
+import mapboxgl from './lib/mapbox';
+import MapboxDirections from './lib/mapbox-directions';
+import MapboxGeocoder from './lib/mapbox-geocoder';
+import { createActions } from './actions';
+import { preparePolygons, applyFog } from './helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function Map({ component, eventHandlers, settings }) {
+  const mapContainerRef = useRef(null);
+  const mapRef = useRef(null);
+
+  const [polygonsArray, setPolygonsArray] = useState([]);
+  const [markersArray, setMarkersArray] = useState([]);
+
+  const { accessToken } = settings;
+
+  const { onMarkerClick, onPolygonClick, onDeterminingGeoposition } = eventHandlers;
+
+  const { markers, polygons, center, fullScreen, navigation, searchBar, geolocation, classList } = component;
+
+  useEffect(() => {
+    mapboxgl.accessToken = accessToken;
+
+    initMapboxLibrary(mapRef, mapContainerRef, component);
+  }, []);
+
+  useEffect(() => {
+    if (mapRef.current && center) {
+
+      mapRef.current.setCenter([center.lng, center.lat]);
+    }
+  }, [center]);
+
+  useEffect(() => {
+    mapRef.current.on('load', () => {
+
+      applyFog(mapRef, component);
+
+      useEvents(mapRef, eventHandlers);
+
+      if (searchBar) {
+        mapRef.current.addControl(
+          new MapboxGeocoder({
+            accessToken: mapboxgl.accessToken,
+            mapboxgl   : mapboxgl,
+          })
+        );
+      }
+
+      if (fullScreen) {
+        mapRef.current.addControl(new mapboxgl.FullscreenControl());
+      }
+
+      if (navigation) {
+        mapRef.current.addControl(new mapboxgl.NavigationControl());
+      }
+
+      if (geolocation) {
+        useGeolocation(mapRef, onDeterminingGeoposition);
+      }
+
+      createActions(mapRef, component);
+    });
+  }, []);
```

Comment:
could we write it all like that

```suggestion
  useEffect(() => {
       initMapboxLibrary(...);
  }, []);
```

No need two useEffect hooks with empty deps